### PR TITLE
Fix issue 420: clarify Facts vs Assertions

### PR DESCRIPTION
**Pull request by gillenhaalb**
Addresses issue #420 with assistance from @cgranade's writing there

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/techniques/testing-and-debugging.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/178a25804a077ffeaaf48608ec62e5101c5e063e/articles/techniques/testing-and-debugging.md) | 80 | 56 | 1 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/eac3e3d6-bda7-4ccb-9331-0cd33b360c7e_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: articles/techniques/testing-and-debugging.md [hunk 1]
```diff
@@ -152,11 +152,32 @@ function AssertPositive(value : Double) : Unit
...
-Thus, if we proceed past a call to `AssertPositive`, we can be assured by that its input was positive.
+Thus, if we proceed past a call to `PositivityFact`, we can be assured by that its input was positive.
+
+Note that we can implement the same behavior as `PositivityFact` using the [`Fact`](xref:microsoft.quantum.diagnostics.fact) function from the <xref:microsoft.quantum.diagnostics> namespace:
+
+```qsharp
+	Fact(value <= 0, "Expected a positive number.");
+```
+
+*Assertions*, on the other hand, are used similarly to facts, but are dependent on the state of the target machine.
```
Comment (cgranade) [inline articles/techniques/testing-and-debugging.md]:
```suggestion
*Assertions*, on the other hand, are used similarly to facts, but may be dependent on the state of the target machine. 
```

File: articles/techniques/testing-and-debugging.md [hunk 2]
```diff
@@ -152,11 +152,32 @@ function AssertPositive(value : Double) : Unit
...
+
+```qsharp
+operation AssertQubitsAreAvailable() : Unit
+{
+     Fact(GetQubitsAvailableToUse() > 0, "No qubits were actually available");
+}
+```
+
+Here, we are using the operation <xref:microsoft.quantum.environment.getqubitsavailabletouse> to return the number of qubits available to use.
+As this clearly depends on the global state of the program and its execution environment, our defined `AssertQubitsAreAvailable` must be an operation as well.
```
Comment (cgranade) [inline articles/techniques/testing-and-debugging.md]:
```suggestion
As this clearly depends on the global state of the program and its execution environment, our definition of  `AssertQubitsAreAvailable` must be an operation as well.
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Lowest<br>category<br>score | Spelling<br>issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/techniques/testing-and-debugging.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/8b0dced39aa92ab65c504c6e2a0ea26f94476f9b/articles/techniques/testing-and-debugging.md) | 80 | 56 | 1 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/baa399d3-1691-4969-b808-482705fa6403_report.html) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Pull request merged**